p2p/transport/quicreuse/connmgr.go

```diff
@@ -131,6 +131,28 @@ func (c *ConnManager) onListenerClosed(key string) {
 	}
 }
 
+func (c *ConnManager) SharedNonQUICPacketConn(network string, laddr *net.UDPAddr) (net.PacketConn, error) {
+	c.quicListenersMu.Lock()
+	defer c.quicListenersMu.Unlock()
+	key := laddr.String()
+	entry, ok := c.quicListeners[key]
+	if !ok {
+		return nil, errors.New("expected to be able to share with a QUIC listener, but no QUIC listener found. The QUIC listener should start first")
+	}
+	t := entry.ln.transport
+	if t, ok := t.(*refcountedTransport); ok {
+		t.IncreaseCount()
```

Comment:
If we move this logic to `quicreuse`, we can do away with the ordering requirement between quic and webrtc listener. 
In quicreuse we'll have to introduce something like `TransportForNonQuicStuff`(hopefully a better name) which is then used for listening on quic.